### DISCUSSION TITLE
Sidebar NavItem OnClick

### DISCUSSION THREAD

**Discussion by curtic00** [vikramlearning/blazorbootstrap#894] [Ideas]
Is this possible in sidebar?

new NavItem { Id = "7",  Href = "#"  ,  IconName = IconName.List, Text = "Toggle Sidebar" , onClick(sidebar.ToggleSidebar()) }

**Comment by raski66-git** [comment]
I would like this feature as well. It would be really nice to get a callback with information on which sidebar item was clicked and the parent hierarchy if it's Sidebar2.

**Comment by ecospecifier** [comment]
Just adding a +1 to this, Doing something OTHER than navigating with clicks from sidebar is a very real use case I have hit on more than one occasion.

Thank your for the awesome work!

**Comment by gvreddy04** [comment]
@curtic00 @raski66-git @ecospecifier Thank you for your feedback. Can you share some real-time scenarios where you want to utilize the onclick feature?

**Comment by curtic00** [comment]
Hi. 

I first thought of this to include a control within the sidebar to collapse/expand the sidebar. 

Another use case would be to launch a modal pop up. 

Sure that there are many others. 

Thanks Vikram, sterling work. 

Charlie 

> Charlie Curtis 
> Worthing, West Sussex
> BN133AB.
> MCSD, MCSA, MCRDBA, Azure administrator
> Mobile: 0777 254 5516
> LinkedIn: https://uk.linkedin.com/in/charlieecharlie

**Comment by raski66-git** [comment]
Hello Vikram,
I love Blazor and Blazor Bootstrap!!! 
My use case - I have a hierarchical data set that displays in a grid at every level, but I would like to have a tree on a sidebar displaying the levels. Once a level is clicked, I would like to display that level's data in the grid.
thanks!
Dave Raskin 

    On Monday, November 11, 2024 at 07:22:45 AM EST, curtic00 ***@***.***> wrote:  
 
 

Hi. 

I first thought of this to include a control within the sidebar to collapse/expand the sidebar. 

Another use case would be to launch a modal pop up. 

Sure that there are many others. 

Thanks Vikram, sterling work. 

Charlie 

> Charlie Curtis 
> Worthing, West Sussex
> BN133AB.
> MCSD, MCSA, MCRDBA, Azure administrator
> Mobile: 0777 254 5516
> LinkedIn: https://uk.linkedin.com/in/charlieecharlie
